client/landing/gutenboarding/stores/domain-suggestions/reducer.ts

```diff
@@ -12,7 +12,7 @@ import * as Actions from './actions';
 import { stringifyDomainQueryObject } from './utils';
 
 const domainSuggestions: Reducer<
-	Record< string, DomainSuggestion[] >,
+	Record< string, DomainSuggestion[] | undefined >,
```

Comment:
Is this accurate? The resolver will now return `[]` if no query is passed.

It's still possible that the selector doesn't find a `DomainSuggestion[]` list for a given key, but that's not relevant here :thinking:

Reply:
Replied in https://github.com/Automattic/wp-calypso/pull/37595/files#r346778273

Reply:
Maybe I didn't reply directly, but that _is_ relevant here, isn't it? This is what lets us know that we may not have a list. It does al hinge on what we decide around my other comment.

Reply:
Well if it can be `undefined` here, we'd need to backtrack all the way through the action and maybe to the resolver. Right now, the [action arg is specified to be `DomainSuggestion[]`](https://github.com/Automattic/wp-calypso/blob/a19c9c57258093b06bd2cc48b6f51f71f1521021/client/landing/gutenboarding/stores/domain-suggestions/actions.ts#L8) -- no `undefined`.

Reply:
813fed0

Reply:
> Well if it can be `undefined` here, we'd need to backtrack all the way through the action and maybe to the resolver. Right now, the [action arg is specified to be `DomainSuggestion[]`](https://github.com/Automattic/wp-calypso/blob/a19c9c57258093b06bd2cc48b6f51f71f1521021/client/landing/gutenboarding/stores/domain-suggestions/actions.ts#L8) -- no `undefined`.

The action creator to receive suggestions shouldn't be passed `undefined` as an argument. These aren't directly related.